llvm/lib/Target/NVPTX/NVPTXInstrInfo.td

```diff
@@ -990,6 +990,34 @@ defm MAD32 : MAD<"mad.lo.s32", i32, B32, i32imm>;
 defm MAD64 : MAD<"mad.lo.s64", i64, B64, i64imm>;
 }
 
+multiclass MAD_WIDE<string PtxSuffix, SDNode Op, ValueType BigVT, NVPTXRegClass BigReg, Operand BigImm, ValueType SmallVT, NVPTXRegClass SmallReg, Operand SmallImm> {
+  def rrr:
+    BasicNVPTXInst<(outs BigReg:$dst),
+              (ins SmallReg:$a, SmallReg:$b, BigReg:$c),
+              "mad.wide." # PtxSuffix,
+              [(set BigVT:$dst, (add (Op SmallVT:$a, SmallVT:$b), BigVT:$c))]>;
+  def rri:
+    BasicNVPTXInst<(outs BigReg:$dst),
+              (ins SmallReg:$a, SmallReg:$b, BigImm:$c),
+              "mad.wide." # PtxSuffix,
+              [(set BigVT:$dst, (add (Op SmallVT:$a, SmallVT:$b), imm:$c))]>;
```

Comment:
I think we should follow the convention of the above MAD cases and only do this lowering when `hasOptEnabled` is true.